starcheck/src/lib/Ska/Starcheck/Obsid.pm

```diff
@@ -1239,6 +1320,44 @@ sub check_star_catalog {
     # store a list of the fid positions
     my @fid_positions = map {{'y' => $c->{"YANG$_"}, 'z' => $c->{"ZANG$_"}}} @{$self->{fid}};
 
+
+    # Make arrays of the items that we need for the hot pixel region check
+    my (@idxs, @yags, @zags, @mags, @types);
+    foreach my $i (1..16){
+        if ($c->{"TYPE$i"} =~ /BOT|GUI|FID/){
+            push @idxs, $i;
+            push @yags, $c->{"YANG$i"};
+            push @zags, $c->{"ZANG$i"};
+            # Add zero to get items that look more like float values in the arrays
+            push @mags, ($c->{"GS_MAG$i"} eq '---') ? 13.94 : $c->{"GS_MAG$i"} + 0.0;
+            push @types, $c->{"TYPE$i"};
+        }
+    }
+
+    # Run the hot pixel region check on the Python side on FID|GUI|BOT
+    my @imposters = check_hot_pix(\@idxs, \@yags, \@zags, \@mags, \@types,
+                                   $self->{ccd_temp}, $self->{date}, $dither_guide_y, $dither_guide_z);
+
+    # Assign warnings based on those hot pixel region checks
+  IMPOSTER:
+    for my $imposter (@imposters){
+        # If the check just fails on the Python side write out a warning and move on.
+        if ($imposter->{status} == 1){
+            push @warn, "$alarm [%d] Processing error when checking for hot pixels.\n";
+            next IMPOSTER;
+        }
+        my $warn = sprintf("$alarm [%2d] Imposter mag %.1f offset %.1f (row % 4d, col % 4d) star (% 4d, % 4d)\n",
+                           $imposter->{idx}, $imposter->{bad2_mag}, $imposter->{offset},
+                           $imposter->{bad2_row}, $imposter->{bad2_col},
+                           $imposter->{entry_row}, $imposter->{entry_col});
+        if ($imposter->{offset} > 4.0){
+            push @warn, $warn;
+        }
+        elsif ($imposter->{offset} > 2.5){
+            push @orange_warn, $warn;
```

Comment:
@taldcroft Your suggestion was critical warn on > 4 and red warn on > 2.5, but critical and red are the same, so I went with orange.